src/Core/src/Platform/Android/HandlerExtensions.cs

```diff
@@ -21,8 +21,7 @@ public static AView ToNative(this IView view, IMauiContext context)
 				if (handler == null)
 					throw new Exception($"Handler not found for view {view}");
 
-				if (handler is IAndroidViewHandler ahandler)
-					ahandler.SetContext(context.Context);
+				handler.SetMauiContext(context);
```

Comment:
This really was a case of not just attaching the Android context to the handler but the whole thing.